test/serializer.test.ts

```diff
@@ -0,0 +1,535 @@
+import t from 'tap'
```

Comment:
Some tests here are testing that X in gives X out. Could we change it to something like:
```
const queries: string[] = [
	"*",
	"@",
....
]

for (const query of queries) {
	t.test('query: ' + query, async (t) => {
    const expected = parse(query)
    const result = serializeString(tree)
    t.strictSame(parse(result), expected)
  })
}
```

makes it easier to add a bunch of queries to test and that we generate the same identical queries, kinda similar to round-trip only that we compare the ast and not the generated string

Reply:
Adding a collection of groq queries derived from our documentation